### ISSUE TITLE
Suggest to modify codebase

### ISSUE THREAD

**Issue by WanrenZeng** [JayParanjape/F-ViTA#6]
Thanks for author's great work! However, there still exists some bugs in the code repo, which makes it difficult to run on my server. I tried to fix them. Hopefully it could be of help.
1. gsam.yml:

This config file contains several dependencies with different cuda version conflict, and many dependencies are not used.
I write a new requirements.txt file with information in gsam.yml.
  `
torch
torchvision
absl-py==2.2.0
accelerate==1.0.1
addict==2.4.0
aiofiles==24.1.0
aiohappyeyeballs==2.4.4
aiohttp==3.10.11
aiosignal==1.3.1
albucore==0.0.17
albumentations==1.4.18
aliyun-python-sdk-core==2.16.0
aliyun-python-sdk-kms==2.16.5
annotated-types==0.7.0
anyio==4.5.2
asttokens==3.0.0
async-timeout==5.0.1
attrs==24.2.0
backcall==0.2.0
beautifulsoup4==4.13.3
bitsandbytes==0.45.0
black==23.12.1
boto3==1.37.13
botocore==1.37.13
bottle==0.13.2
cachetools==5.5.2
cffi==1.17.1
ci-info==0.3.0
clean-fid==0.1.35
colorama==0.4.6
coloredlogs==15.0.1
comm==0.2.2
common==0.1.2
configobj==5.0.9
configparser==7.1.0
contourpy==1.1.1
crcmod==1.7
cryptography==44.0.0
cycler==0.12.1
cython==3.0.12
data==0.4
datasets==3.1.0
debugpy==1.8.9
decorator==5.1.1
decord==0.6.0
defusedxml==0.7.1
diffusers==0.31.0
dill==0.3.8
distro==1.9.0
docker-pycreds==0.4.0
dual==0.0.10
dynamo3==0.4.10
easydict==1.13
et-xmlfile==2.0.0
eval-type-backport==0.2.0
exceptiongroup==1.2.2
executing==2.1.0
fairscale==0.4.4
filelock==3.14.0
fire==0.7.0
fitz==0.0.1.dev2
flatbuffers==24.3.25
flywheel==0.5.4
fonttools==4.55.2
frontend==0.0.3
frozenlist==1.5.0
fsspec==2024.9.0
ftfy==6.2.3
funcsigs==1.0.2
gitdb==4.0.11
gitpython==3.1.43
google-auth==2.38.0
google-auth-oauthlib==1.0.0
grpcio==1.70.0
h11==0.14.0
httpcore==1.0.7
httplib2==0.22.0
httpx==0.27.2
huggingface-hub==0.26.5
humanfriendly==10.0
imageio==2.35.1
importlib-metadata==8.5.0
importlib-resources==6.4.5
ipykernel==6.29.5
ipython==8.12.3
isodate==0.6.1
isort==5.13.2
itsdangerous==2.2.0
jedi==0.19.2
jiter==0.8.2
jmespath==0.10.0
joblib==1.4.2
jsonschema==4.23.0
jsonschema-specifications==2023.12.1
jupyter-client==8.6.3
jupyter-core==5.7.2
kiwisolver==1.4.7
lazy-loader==0.4
lmdb==1.6.2
looseversion==1.3.0
lpips==0.1.4
lxml==5.3.0
markdown==3.7
markdown-it-py==3.0.0
matplotlib==3.7.5
matplotlib-inline==0.1.7
mdurl==0.1.2
model-index==0.1.11
multidict==6.1.0
multiprocess==0.70.16
mypy-extensions==1.0.0
nest-asyncio==1.6.0
nibabel==5.2.1
nipype==1.8.6
numpy==1.24.4
oauthlib==3.2.2
onnx==1.17.0
onnxruntime==1.19.2
openai==1.57.1
opencv-contrib-python==4.11.0.86
opencv-python==4.10.0.84
opencv-python-headless==4.11.0.86
ordered-set==4.1.0
oss2==2.17.0
packaging==24.2
paddle==1.0.2
paddleocr==2.10.0
pandas==2.0.3
parso==0.8.4
pathlib==1.0.1
pathspec==0.12.1
pdf2image==1.17.0
pdfminer-six==20231228
pdfplumber==0.11.5
peewee==3.17.9
peft==0.13.2
pexpect==4.9.0
pickleshare==0.7.5
pkgutil-resolve-name==1.3.10
platformdirs==4.3.6
positional-encodings==6.0.3
prompt-toolkit==3.0.48
propcache==0.2.0
protobuf==5.29.1
prov==2.0.1
prox==0.0.17
psutil==6.1.0
ptyprocess==0.7.0
pure-eval==0.2.3
py-cpuinfo==9.0.0
pyarrow==17.0.0
pyasn1==0.6.1
pyasn1-modules==0.4.1
pyclipper==1.3.0.post6
pycocoevalcap==1.2
pycocotools==2.0.7
pycparser==2.22
pycryptodome==3.21.0
pydantic==2.10.3
pydantic-core==2.27.1
pydot==3.0.4
pygments==2.18.0
pymupdf==1.24.11
pyparsing==3.1.4
pypdf2==3.0.1
pypdfium2==4.30.1
pytesseract==0.3.13
python-dateutil==2.9.0.post0
python-docx==1.1.2
python-dotenv==1.0.1
python-geoip-python3==1.3
pytz==2023.4
pywavelets==1.4.1
pyxnat==1.6.2
pyzmq==26.2.0
rapidfuzz==3.9.7
rdflib==6.3.2
referencing==0.35.1
regex==2024.11.6
requests-oauthlib==2.0.0
rich==13.4.2
rpds-py==0.20.1
rsa==4.9
ruff==0.0.259
s3transfer==0.11.4
safetensors==0.4.5
scikit-image==0.21.0
scikit-learn==1.3.2
scipy==1.10.0
seaborn==0.13.2
sentry-sdk==2.19.2
setproctitle==1.3.4
setuptools==60.2.0
shapely==2.0.7
simplejson==3.19.3
six==1.17.0
smmap==5.0.1
sniffio==1.3.1
soupsieve==2.6
stack-data==0.6.3
starlette==0.44.0
submitit==1.5.2
supervision==0.25.0
tabulate==0.9.0
tensorboard==2.14.0
tensorboard-data-server==0.7.2
termcolor==2.4.0
threadpoolctl==3.5.0
tifffile==2023.7.10
tight==0.1.0
tiktoken==0.7.0
timm==0.3.2
tokenizers==0.20.3
tomli==2.2.1
tornado==6.4.2
tqdm==4.67.1
traitlets==5.14.3
traits==6.3.2
transformers==4.46.3
typing-extensions==4.12.2
tzdata==2024.2
ultralytics==8.3.85
ultralytics-thop==2.0.14
urllib3==1.26.20
uvicorn==0.33.0
wandb==0.19.0
wcwidth==0.2.13
werkzeug==3.0.6
xxhash==3.5.0
yapf==0.43.0
yarl==1.15.2
zipp==3.20.2`

Then install Grounded-Segment-Anything, recognize-anything and grounded-segment-anything-osx

2. inference_gsam.py
`images, gts, names = get_val_images(dataset_name)
`
The get_val_images function seems not defined?
I write a inference.py with single image.
`import argparse
import logging
import math
import os

import PIL.Image
# os.environ['HF_DATASETS_CACHE']="/mnt/store/jparanj1/.cache/"
# os.environ['TRANSFORMERS_CACHE']='/mnt/store/jparanj1/.cache/'
# os.environ['HF_HOME']="/mnt/store/jparanj1/.cache/"
# os.environ['HF_HUB_CACHE']='/mnt/store/jparanj1/.cache/'
os.environ["HF_ENDPOINT"]="https://hf-mirror.com"
from pathlib import Path
from typing import Optional

import accelerate
import datasets
import diffusers
import numpy as np
import PIL
import requests
import torch
import torch.nn.functional as F
import torch.utils.checkpoint
import transformers
from accelerate import Accelerator
from accelerate.logging import get_logger
from accelerate.utils import ProjectConfiguration, set_seed
from datasets import load_dataset
from diffusers import (AutoencoderKL, DDPMScheduler,
                       StableDiffusionInstructPix2PixPipeline,
                       UNet2DConditionModel)
from diffusers.optimization import get_scheduler
from diffusers.training_utils import EMAModel
from diffusers.utils import check_min_version, deprecate, is_wandb_available
from diffusers.utils.import_utils import is_xformers_available
from huggingface_hub import HfFolder, Repository, create_repo, whoami
from packaging import version
from torchvision import transforms
from tqdm.auto import tqdm
from transformers import CLIPTextModel, CLIPTokenizer
from diffusers import StableDiffusionInstructPix2PixPipeline
from gsampipeline import StableDiffusionInstructPix2PixGSAMPipeline
from diffusers.utils import make_image_grid, load_image
from utils import *

# pretrained_model_name_or_path = sys.argv[1]
# save_name = sys.argv[2]
# dataset_name = sys.argv[3]
# pretrained_model_name_or_path = "jay-jnp/F-VITA_FLIR"
pretrained_model_name_or_path = "/path/to/F-ViTA_FLIR"
save_path = "./output"
num_inference_steps = 100
return_text = True
return_boxes = False
return_masks = True
# Load scheduler, tokenizer and models.

unet = UNet2DConditionModel.from_pretrained(
    pretrained_model_name_or_path,
    subfolder="unet",
    revision=None,
)

ema_unet = EMAModel(
            unet.parameters(), model_cls=UNet2DConditionModel, model_config=unet.config
)

im = "/home/zengpeiyi/work_base/datasets/dreambooth-dataset/grey_sloth_plushie/00.jpg"
#added linear projection
added_linear = torch.nn.Linear(1024, 768)
added_linear.load_state_dict(torch.load(os.path.join(pretrained_model_name_or_path,'added_linear.pth')))
pipeline = StableDiffusionInstructPix2PixGSAMPipeline(
    unet = unet,
    added_linear = added_linear,
    return_boxes=return_boxes,
    return_masks=return_masks,
    return_text=return_text
)

#get list of images for validation
# images, gts, names = get_val_images(dataset_name)
# preds = []
# save_path = os.path.join("predictions", dataset_name, save_name)
os.makedirs(save_path, exist_ok=True)
name = os.path.basename(im)
init_image = load_image(im)
with torch.autocast(
                    'cuda',
                    enabled=True,
):
    pred_im = pipeline("Create an infrared version of the given image. Make it long wave infrared", image=init_image, im_path=im, num_inference_steps=num_inference_steps).images[0]

    pred_im.save(os.path.join(save_path, name))
    # preds.append(os.path.join(save_path, names[i]))`

3. gsampipeline.py
In gsampipeline.py, when loading sam vit, there exists a error. The repo recommend checkpoint from https://huggingface.co/datasets/Gourieff/ReActor/blob/main/models/sams/sam_vit_b_01ec64.pth , but sam_vit_h is tried to be loaded.
in
F-ViTA/Grounded-Segment-Anything/segment_anything/segment_anything/build_sam.py
delete
`build_sam = build_sam_vit_h
`
and add 
`build_sam = build_sam_vit_b
`
before 
`
sam_model_registry = {
    "default": build_sam,
    "vit_h": build_sam,
    "vit_l": build_sam_vit_l,
    "vit_b": build_sam_vit_b,
}`
that is:
`
build_sam = build_sam_vit_b
sam_model_registry = {
    "default": build_sam,
    "vit_h": build_sam,
    "vit_l": build_sam_vit_l,
    "vit_b": build_sam_vit_b,
}`

4. checkpoint download:
Recommend:
`huggingface-cli download --resume-download repo_id_for_FViTA
`
instead of directly download checkpoints by from_pretrained(), as it may throw errors because of lack of "added_linear.pth"
